instrumentation/rake/lib/opentelemetry/instrumentation/rake/instrumentation.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rake
+      # The Instrumentation class contains logic to detect and install the Rake instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('0.9.0')
+
+        install do |_config|
+          require_dependencies
+          patch_rake
+          shutdown_at_exit
+        end
+
+        present do
+          defined?(::Rake::Task)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        private
+
+        def gem_version
+          Gem::Version.new(::Rake::VERSION)
+        end
+
+        def require_dependencies
+          require_relative './patches/task'
+        end
+
+        def patch_rake
+          ::Rake::Task.prepend(Patches::Task)
+        end
+
+        def shutdown_at_exit
+          return unless defined?(::Rake) && !::Rake.application.top_level_tasks.empty?
+
+          at_exit { OpenTelemetry.tracer_provider.shutdown }
```

Comment:
Thanks for your feedback @arielvalentin !
I'll wait for their response before making any change here.

Reply:
A similar situation occurred in the ActiveJob instrumentation (although that was a `force_flush` and not a shutdown).

1. Maybe force_flush is a more palatable option?
2. We decided to hide that behind a configuration flag: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/patches/active_job_callbacks.rb#L63

Reply:
I took another look at the datadog instrumentation and it is calling `shutdown` in the ensure blocks:
https://github.com/DataDog/dd-trace-rb/blob/v1.1.0/lib/datadog/tracing/contrib/rake/instrumentation.rb#L29

I made a change to call `force_flush` in the eusure blocks instead of adding an `at_exit` hook: cba89c36548678c10f2a0da599a87b484996319b

how is this?

Reply:
@arielvalentin I have updated this PR to not use `at_exit` hook but let me know if there is anything I need to change. Thanks!!